xbmc/ServiceBroker.cpp

```diff
@@ -248,9 +248,16 @@ CDatabaseManager& CServiceBroker::GetDatabaseManager()
   return g_application.m_ServiceManager->GetDatabaseManager();
 }
 
-CEventLog& CServiceBroker::GetEventLog()
+CEventLog* CServiceBroker::GetEventLog()
 {
-  return g_serviceBroker.m_pSettingsComponent->GetProfileManager()->GetEventLog();
+  if (g_serviceBroker.m_pSettingsComponent)
```

Comment:
Maybe use negative check here to reduce the nested if statements.

Reply:
id end up having to do a return nullptr, and then still have the fallthrough return nullptr.
Is there any benefit?

```
CEventLog* CServiceBroker::GetEventLog()
{
  if (!g_serviceBroker.m_pSettingsComponent)
    return nullptr;
  
  auto profileManager = g_serviceBroker.m_pSettingsComponent->GetProfileManager();
  if (profileManager)
    return &profileManager->GetEventLog();
  
  return nullptr;
}
```

Reply:
What about this?

```
CEventLog* CServiceBroker::GetEventLog()
{
  if (!g_serviceBroker.m_pSettingsComponent)
    return nullptr;
  
  auto profileManager = g_serviceBroker.m_pSettingsComponent->GetProfileManager();
  if (!profileManager)
    return nullptr;

  return &profileManager->GetEventLog();
}

```

Reply:
Sure, im easy.